oeps/architectural-decisions/oep-0065/decisions/0002-frontend-app-migrations.rst

```diff
@@ -0,0 +1,120 @@
+
+The shell will be configured to load things like "courseware" or "login", not "learning" and "authn" - the repository the module is in is incidental.
+
+Regarding "plugins"
+-------------------
+
+Plugins are another type of module.  We believe that they are different than application modules because they have a different interface with the shell.  Plugins are generally UI widgets loaded into a slot, though some can be larger.  Applications are generally large and complex, comprising multiple pages of functionality with internal routing.  They are also loaded at a path in the application, not in a plugin slot.
+
+Modeling applications as Plugins is a possibility, but we believe that it feels like too much indirection and abstraction, and minimizes the importance of applications.  Without them there's no site, and many are likely *required* functionality for the Open edX frontend to function at all.
+
+Some frontend-app-* repositories may also include Plugins, but they aren't the primary subject of this ADR.
+
+Consequences
+************
+
+As the module architecture stabilizes, frontend-app-* maintainers and developers will be encouraged to migrate their micro-frontends into application module libraries, and to adopt the module architecture provided by frontend-base.  (There will be a migration guide.)
+
+As micro-frontends are migrated application modules using the shell, there will be a deployment approach that mimics the micro-frontend architecture, but which will require operators to adopt a new underlying configuration and build process to achieve a similar result.  Each frontend-app-* repository will need a deprecation process for the micro-frontend configuration and build infrastructure.
+
+Unsupported Customizations
+==========================
+
+The micro-frontend architecture took an extreme approach to "flexibility", allowing MFEs to diverge from each other in a variety of ways as described in OEP-65.  As a result, in the process of migrating them to the module architecture, there could be unforeseen refactoring that may need to happen in some MFEs that don't map into modules well, or which have customizations that aren't supported by the Shell.  While we hope to provide enough extensibility mechanisms to reduce the need for forking or hacky customizations, there will be customizations we haven't anticipated, which the community will need to work around or find ways to support.
+
+Consistent Dependency Versions
+==============================
+
+The shell will support specific versions of shared dependencies (such as React, Paragon, or React Router).  All applications loaded into the shell's Site will be expected to use (or at least be compatible) with that version.  We intend to create lock-step version consistency of shared dependencies across all applications in the platform.  We envision each Open edX release supporting a particular major version of each shared dependency.
```

Comment:
It might be nice to note that the *lack* of consistent dependency versions has been a pain point in the MFE architecture.